sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -266,8 +266,26 @@ case class Invoke(
   override def nullable: Boolean = targetObject.nullable || needNullCheck || returnNullable
   override def children: Seq[Expression] = targetObject +: arguments
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  override def eval(input: InternalRow): Any = {
+    val obj = targetObject.eval(input)
+    val args = arguments.map(e => e.eval(input).asInstanceOf[Object])
+    val argClasses = CallMethodViaReflection.expressionJavaClasses(arguments)
+    val method = obj.getClass.getDeclaredMethod(functionName, argClasses : _*)
+    if (needNullCheck && args.exists(_ == null)) {
+      // return null if one of arguments is null
+      null
+    } else {
+      val ret = method.invoke(obj, args: _*)
+
+      if (CodeGenerator.defaultValue(dataType) == "null") {
+        ret
+      } else {
+        // cast a primitive value using Boxed class
+        val boxedClass = CallMethodViaReflection.typeBoxedJavaMapping(dataType)
+        boxedClass.cast(ret)
```

Comment:
Do we need to do explicitly cast? Doesn't Scala cast it if it's needed?

Reply:
It would work without this cast in Scala. I tried to follow [this cast|https://github.com/kiszk/spark/blob/d81c9502a94b30a13afe284fc7e389edd9da02a0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala#L336] that may throw an exception.

Reply:
For interpreted execution, I think it is less meaningful to check that. cc @hvanhovell

Reply:
The point is where we want to cause a cast exception. With this code, an exception will occur at this class. Without this code, an exception will occur anywhere in the code.